src/core/thread/topology.cpp

```diff
@@ -121,6 +121,25 @@ otError Child::RemoveIp6Address(const Ip6::Address &aAddress)
     return error;
 }
 
+otError Child::GetIp6AddressWithPrefix(const Ip6::Address &aPrefix, uint8_t aPrefixLen, Ip6::Address &aAddress)
+{
+    otError error = OT_ERROR_NOT_FOUND;
+
+    for (uint16_t index = 0; index < kMaxIp6AddressPerChild; index++)
+    {
+        VerifyOrExit(!mIp6Address[index].IsUnspecified());
+
+        if (memcmp(mIp6Address[index].mFields.m8, aPrefix.mFields.m8, aPrefixLen) == 0)
```

Comment:
suggestion: how about using `Ip6::Address::PrefixMatch()` here to check the match?

Reply:
We can probably remove this method with #2504.